### DISCUSSION TITLE
Hierarchical component

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#1238] [Q&A (Help)]
A client has asked me to create a hierarchical data visualization. Specifically, the data needs to be arranged to display the service type as the root of the processes. The structure should look like this:

```
Service Type 1
    Process 1
    Process 2
    Process 3

Service Type 2
    Process 4
    Process 5
    Process 6
```

Could someone explain how I could create this kind of visualization in Laravel Backpack? The grid should also have filters so that users can customize the interface and data. Please let me know; this is very urgent.

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment]
Hey @cod3rshotout 

Are you looking for [Reorder Operation](https://backpackforlaravel.com/docs/6.x/crud-operation-reorder)? This lets admin to reorder & nest entries. 

![image](https://github.com/user-attachments/assets/92f1fef5-8e99-40bc-a3d7-fb38e436b47a)

**Reply by cod3rshotout** [reply]
Hi @karandatwani92 thanks for the answer, I'm looking for a hierarchical datatable, something like this:
https://stackoverflow.com/questions/52039222/how-to-implement-hierarchical-multilevel-datatable-in-javascript
https://community.dataiku.com/discussion/15831/improve-user-experience-for-hierarchical-data

I simply need as a root element the "service type", and when the user click a service type then the other rows appear as nested childs.

**Comment by karandatwani92** [comment]
Hey @cod3rshotout 

- Backpack supports Datatable's [row_details](https://datatables.net/examples/api/row_details.html) feature. It is documented [here](https://backpackforlaravel.com/docs/6.x/crud-operation-list-entries#details-row-1).
- if you are looking for [row_grouping](https://datatables.net/examples/advanced_init/row_grouping.html). This does not come out of the box.

**Reply by cod3rshotout** [reply]
@karandatwani92 hi, sorry for late reply .. thanks for your response, anyway could you suggest the best approach to replicate this using backpack: https://stackoverflow.com/questions/18274826/sorting-table-with-hierarchical-data
or maybe there is a custom addon to do that? thanks

**Reply by karandatwani92** [reply]
Right now, I think of method overriding,  modifying ListOperation `API` response and `list` view according to the requirement.

I can imagine myself creating a copy of the list view blade file, modifying its datatables config and overriding list operations methods in the controller:
- overriding `index()` method to pass the custom list view.
- overriding `search()` method to modify API response.

**Reply by cod3rshotout** [reply]
@karandatwani92 thanks for the answer and happy new year! is there a way to contact you directly? I would like to know if you can offer me support in the development of this component (for a fee of course)
